caikit/core/module_backends/base.py

```diff
@@ -66,3 +69,21 @@ def stop(self):
     def start_lock(self):
         with self._start_lock:
             yield
+
+    def handle_runtime_context(  # noqa: B027
+        self,
+        model_id: str,
+        runtime_context: RuntimeServerContextType,
```

Comment:
Looks like there's an assumption baked in here that the information in any given context will be the same for the same model id. If multiple concurrent requests come in for the same model this won't be able to disambiguate, right?

Seems fine for the use case where a given model just needs some extra metadata to be loaded and it'll be the same on every request, but not so fine if somebody tries to use this to add tracing info to inference calls

Reply:
Ah, yeah, that is definitely true. The `context_arg` stuff that @HonakerM added is the right way to do per-call tracing (WIP at the moment).

Reply:
>If multiple concurrent requests come in for the same model this won't be able to disambiguate, right?

Yes, this would definitely fall to the backend itself to handle